api/lib/models/admin/admin-data-mapper-impl.ts

```diff
@@ -0,0 +1,197 @@
+import * as admin from 'firebase-admin';
+import { Inject, Injectable } from 'injection-js';
+import { from } from 'rxjs';
+import { map } from 'rxjs/operators';
+import squel from 'squel';
+import tsStream from 'ts-stream';
+import { IEmail } from '.';
+import { UidType } from '../../JSCommon/common-types';
+import { HttpError, MethodNotImplementedError } from '../../JSCommon/errors';
+import { ICustomPermissions } from '../../services/auth';
+import { AuthLevel, IAuthService } from '../../services/auth/auth-types';
+import { IAcl, IAdminAclPerm } from '../../services/auth/RBAC/rbac-types';
+import { IEmailService } from '../../services/communication/email';
+import { IDbResult } from '../../services/database';
+import { GenericDataMapper } from '../../services/database/svc/generic-data-mapper';
+import { MysqlUow } from '../../services/database/svc/mysql-uow.service';
+import { IUowOpts } from '../../services/database/svc/uow.service';
+import { IAdminDataMapper } from './index';
+import { EmailHistory } from './types/email-history';
+import UserRecord = admin.auth.UserRecord;
+
+@Injectable()
+export class AdminDataMapperImpl extends GenericDataMapper implements IAdminDataMapper,
+                                                                      IAdminAclPerm {
+  public GET_EMAIL: string = 'admin:get_email';
+  public SEND_EMAIL: string = 'admin:send_email';
+  public CREATE: string = 'admin:create';
+  public REDUCE_PERMISSION: string = 'admin:reduce_perm';
+  public MAKE_ACTIVE: string = 'admin:make_active';
+  // Undefined actions for admin mapper
+  public COUNT: string;
+  public DELETE: string;
+  public READ: string;
+  public READ_ALL: string;
+  public UPDATE: string;
+  public tableName: string;
+
+  // Undefined properties for admin mapper;
+  protected pkColumnName: string;
+
+  constructor(
+    @Inject('IAcl') readonly acl: IAcl,
+    @Inject('IAuthService') private readonly authService: IAuthService,
+    @Inject('MysqlUow') private sql: MysqlUow,
```

Comment:
Is this supposed to be protected and readonly?

Reply:
yep. good catch